RabbitMQ.Stream.Client/StreamSystem.cs

```diff
@@ -121,17 +133,32 @@ public async Task<bool> StreamExists(string stream)
         /// <returns></returns>
         public async Task<ulong> QueryOffset(string reference, string stream)
         {
-            if (string.IsNullOrEmpty(reference))
-            {
-                return 0;
-            }
+            MaybeThrowQueryException(reference, stream);
+
+            var response = await client.QueryOffset(reference, stream);
+            ClientExceptions.MaybeThrowException(response.ResponseCode,
+                $"QueryOffset stream: {stream}, reference: {reference}");
+            return response.Offset;
+        }
+
+        /// <summary>
+        /// QueryOffsetWithDefaultValue retrieves the last consumer offset stored
+        /// given a consumer name and stream name 
+        /// or 0 as default value if thre is no stored value
+        /// </summary>
+        /// <param name="reference">Consumer name</param>
+        /// <param name="stream">Stream name</param>
+        /// <returns></returns>
+        public async Task<ulong> QueryOffsetWithDefaultValue(string reference, string stream)
```

Comment:
I would add an overload for `QueryOffset(string reference, string stream, ulong defaultValue)`